addons/mail/static/src/messaging/entity/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -123,6 +130,46 @@ function MessagingNotificationHandlerFactory({ Entity }) {
             }
         }
 
+        /**
+         * @private
+         * @param {Object} param0
+         * @param {integer} param0.channelId
+         * @param {integer} param0.id
+         * @param {integer} param0.last_message_id
+         * @param {integer} param0.partner_id
+         */
+        async _handleNotificationChannelFetched({
+            id,
+            channelId,
+            last_message_id,
+            partner_id,
+        }) {
+            const channel = this.env.entities.Thread.find(thread =>
+                thread.id === channelId && thread.model === 'mail.channel'
+            );
+            if (!channel) {
+                // create channel ?
```

Comment:
Probably not, supposedly when the channel will be "known" for an explicit reason (eg. displayed on a threadviewer), the seen info should come with fetched data anyway.